React sample integrating with Azure SDKs (#12924)
Azure/azure-sdk-for-js: samples/frameworks/react/ts/README.md

# Azure SDK samples for React (TypeScript)

This sample application shows how to use the TypeScript client libraries for Azure in some common scenarios.

In this sample, we build a simple Todo application in React using [create-react-app][react] and integrating with various Azure services.

- Integration with Azure EventHubs showcases a mechanism for multiple instances to consume EventHub messages produced by any one of them.
- Integration with Azure ServiceBus showcases a mechanism for any single instance to consume ServiceBus messages. When multiple windows are open, only one of them will receive any given message.
- Integration with Azure Storage Blob showcases a mechanism for fetching Azure blobs and displaying them in the browser.

## Prerequisites

The samples are compatible with Node.js >= 8.0.0.

Before running the samples in Node, they must be compiled to JavaScript using the TypeScript compiler. For more information on TypeScript, see the [TypeScript documentation][typescript].

You need [an Azure subscription][freesub] and the following resources created to run this sample:

> - An Azure EventHubs namespace. Please refer to the [EventHubs documentation][eventhubs] for additional information on EventHubs

Reviewer: In case anyone else runs into this.. 

This was caused because I had connection string in the env variable when the expected was the namespace endpoint.
`<namespace>.servicebus.windows.net`.

> - An Azure EventHubs namespace. Please refer to the [EventHubs documentation][eventhubs] for additional information on EventHubs

Reviewer: Thanks @maorleger

---
page_type: sample
languages:
  - typescript
products:
  - azure
  - azure-event-hubs
  - azure-service-bus
  - azure-storage
urlFragment: react-typescript
---

# Azure SDK samples for React (TypeScript)

This sample application shows how to use the TypeScript client libraries for Azure in some common scenarios.

> In this sample, we build a simple Todo application in React using [create-react-app][react] and integrating with various Azure services

Reviewer: ```suggestion
In this sample, we build a simple Todo application in React using [create-react-app][react] and integrating with various Azure services.
```

- Integration with Azure EventHubs showcases a mechanism for multiple instances to consume EventHub messages produced by any one of them.
- Integration with Azure ServiceBus showcases a mechanism for any single instance to consume ServiceBus messages. When multiple windows are open, only one of them will receive any given message.
- Integration with Azure Storage Blob showcases a mechanism for fetching Azure blobs and displaying them in the browser.

> ## Prerequisites

Author: Documentation is hard! I'd love to hear about where I can improve the docs. Especially interested in how to make the README better. There's a non-trivial amount of setup and permission management that I did not know how to properly capture.

The samples are compatible with Node.js >= 8.0.0.

Before running the samples in Node, they must be compiled to JavaScript using the TypeScript compiler. For more information on TypeScript, see the [TypeScript documentation][typescript]. Install the TypeScript compiler using

```bash
npm install -g typescript
```

> You need [an Azure subscription][freesub] and the following resources created to run this sample:

Reviewer 2: @sadasant knows the magic you need to insert into your readme to get one of those "deploy to azure" buttons. Perhaps that would help? If so, then all of these test resources can be put into a single ARM template.

> You need [an Azure subscription][freesub] and the following resources created to run this sample:

Author: Thank you both for the feedback! I added an ARM template, and my plan is to expand the setup instructions here for things that cannot be templated. So stay tuned - updates will be coming. I'll resolve this conversation when I have finished making changes

- An Azure EventHubs namespace. Please refer to the [EventHubs documentation][eventhubs] for additional information on EventHubs
- An Azure ServiceBus namespace and queue. Please refer to the [ServiceBus documentation][servicebus] for additional information on ServiceBus.
- An Azure Storage Blob container, with a single text file uploaded called "todo.txt" to support the sample. Please refer to the [Storage Blob documentation][storageblob] for additional information on Azure Storage Blob. This file will be fetched from Azure Storage Blob and displayed on the screen.
- Finally, you'll need a way to authenticate the application with Azure. This requires some additional setup configuring the correct access permissions to the above resources using either a service principal or a role-based-authentication. Please refer to the [@azure/identity][identity] package for information on authentication.

## Running the sample

Once the above created you'll want to ensure React has the necessary environment variables. To do this, copy `sample.env` as `.env` and provide the necessary environment variables to configure the application.

Run the sample app:

```base
npm start
```

> A new browser window will open containing a sample set of Todos that have been created. Since this is a contrived example you can only create new Todos and complete existing Todos. As you interact with the application you'll notice EventHubs and ServiceBus messages get written out to the console in the the browser's developer tools.

Reviewer: ```suggestion
A new browser window will open containing a sample set of Todos that have been created. Since this is a contrived example, you can only create new Todos and complete existing Todos. As you interact with the application, you'll notice EventHubs and ServiceBus messages get written out to the console in the browser's developer tools.
```

> Additionally you may open multiple instances of this sample application and watch as Todos synchronize in real time.

Reviewer: ```suggestion
Additionally, you may open multiple instances of this sample application and watch as Todos synchronize in real-time.
```

# Azure SDK samples for React (TypeScript)

This sample application shows how to use the TypeScript client libraries for Azure in some common scenarios.

In this sample, we build a simple Todo application in React using [create-react-app][react] and integrating with various Azure services.

> - Integration with Azure EventHubs showcases a mechanism for multiple instances to consume EventHub messages produced by any one of them.

Reviewer 3: Maybe we can reword this to _why_ we're using these services. Something like
```suggestion
- Integration with Azure Event Hubs to support real-time updates across multiple instances of the application.
- Integration with Azure Storage Blob for persisting ToDo items.
```

- Integration with Azure Storage Blob showcases a mechanism for fetching Azure blobs and displaying them in the browser.

## Prerequisites

The samples are compatible with Node.js >= 8.0.0.

Before running the samples in Node, they must be compiled to JavaScript using the TypeScript compiler. For more information on TypeScript, see the [TypeScript documentation][typescript].

You need [an Azure subscription][freesub] and the following resources created to run this sample:

> - An Azure EventHubs namespace. Please refer to the [EventHubs documentation][eventhubs] for additional information on EventHubs.

Reviewer 3: ```suggestion
- An Azure Event Hubs namespace. Please refer to the [Event Hubs documentation][eventhubs] for additional information on Event Hubs.
```

# Azure SDK samples for React (TypeScript)

This sample application shows how to use the TypeScript client libraries for Azure in some common scenarios.

In this sample, we build a simple Todo application in React using [create-react-app][react] and integrating with various Azure services.

- Integration with Azure Event Hubs to support real-time updates across multiple instances of the application.
- Integration with Azure Storage Blob for persisting ToDo items.

> - Integration with Azure Storage Blob showcases a mechanism for fetching Azure blobs and displaying them in the browser.

Reviewer 3: ```suggestion
```
This line can be removed since the earlier suggestion was merged.
